Add WordPress

Mytraas opened the pull request:
## Pull Request Summary

### Docker Compose YAML File (`docker-compose.yml`)

1. WordPress image version `6.4.1`.
2. Utilized environment variables for configuration:
   - `WORDPRESS_DB_HOST` set to `mariadb`.
   - `WORDPRESS_DB_NAME` set using `${DB_NAME}`.
   - `WORDPRESS_DB_USER` set using `${DB_USER}`.
   - `WORDPRESS_DB_PASSWORD` set using `${DB_PASSWORD}`.
3. Defined volumes to persist WordPress data in `${APP_DATA_DIR}/data/wordpress:/var/www/html`.
4. Mapped the container port `${APP_PORT}` to port `80`.
5. Added Traefik labels for routing and HTTPS configuration.
6. Included dependencies on the `mariadb` service.
7. Specified the network as `tipi_main_network`.

### Docker Compose Service (`mariadb`)

1. MariaDB image version `11.1.3`.
2. Used environment variables for MariaDB configuration:
   - `MYSQL_ROOT_PASSWORD` set using `${DB_ROOT_PASSWORD}`.
   - `MYSQL_DATABASE` set using `${DB_NAME}`.
   - `MYSQL_USER` set using `${DB_USER}`.
   - `MYSQL_PASSWORD` set using `${DB_PASSWORD}`.
3. Defined volumes to persist MariaDB data in `${APP_DATA_DIR}/data/mariadb:/var/lib/mysql`.
4. Specified the network as `tipi_main_network`.

### JSON Configuration for WordPress

| Key                          | Value                              |
|------------------------------|------------------------------------|
| "name"                       | "WordPress"                        |
| "available"                  | true                               |
| "port"                       | 8213                               |
| "exposable"                  | true                               |
| "id"                         | "wordpress"                        |
| "description"                | "WordPress is a popular content management system for creating websites and blogs." |
| "tipi_version"               | 1                                  |
| "version"                    | "6.4.1"                            |
| "categories"                 | ["development"]                    |
| "short_desc"                 | "Popular CMS for websites and blogs" |
| "author"                     | "WordPress.org"                    |
| "source"                     | "https://github.com/WordPress/WordPress" |
| "website"                    | "https://wordpress.org"            |
| "form_fields"                | Array of form field objects        |
| "supported_architectures"    | ["amd64", "arm64"]                 |

steveiliop56 commented on apps/wordpress/config.json:
```diff
@@ -0,0 +1,45 @@
+  "source": "https://github.com/WordPress/WordPress",
+  "website": "https://wordpress.org",
+  "form_fields": [
+    {
+      "type": "text",
+      "label": "Database Name",
+      "required": true,
+      "env_variable": "DB_NAME"
+    },
+    {
+      "type": "text",
+      "label": "Database User",
+      "required": true,
+      "env_variable": "DB_USER"
+    },
+    {
+      "type": "password",
+      "label": "Database Password",
+      "required": true,
+      "env_variable": "DB_PASSWORD"
+    },
+    {
+      "type": "password",
+      "label": "Database Root Password",
+      "required": true,
+      "env_variable": "DB_ROOT_PASSWORD"
+    }
+  ],
```
You don't need to ask user to add db details. Please use the random function instead.

steveiliop56 commented on apps/wordpress/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+        traefik.http.routers.wordpress-web-local.service: wordpress-web
+        traefik.http.routers.wordpress-web-local.tls: true
+
+  mariadb:
+    container_name: mariadb
```
Please rename that to something like ```wordpress-db``` or ```wordpress-mariadb```.

steveiliop56 commented on apps/wordpress/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+      - MYSQL_PASSWORD=${DB_PASSWORD}
+    volumes:
+      - ${APP_DATA_DIR}/data/mariadb:/var/lib/mysql
+    restart: always
```
It would be better to use ```unless-stopped```.

steveiliop56 commented on apps/wordpress/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+      - ${APP_DATA_DIR}/data/wordpress:/var/www/html
+    ports:
+      - ${APP_PORT}:80
+    restart: always
```
It would be better to use ```unless-stopped```

steveiliop56 commented on apps/wordpress/config.json:
```diff
@@ -0,0 +1,45 @@
+  "description": "WordPress is a popular content management system for creating websites and blogs.",
+  "tipi_version": 1,
+  "version": "6.4.1",
+  "categories": ["development"],
```
I think it would be better to use the social instead of development.

steveiliop56 reviewed (changes requested):
Hello @Mytraas,

Thank you for the pr! I left you some notes for small things to fix and then we can merge!

Mytraas commented:
I've implemented all the requested changes in my latest commit. Below, you can find a detailed overview of the modifications I've made:

## Summary of Recent Updates

### JSON Configuration (`config.json`)

- Updated "categories" to ["social"] (previously "development").
- Updated "form_fields" with "type": "random" and "min": 32 for enhanced security.

### Docker Compose YAML File (`docker-compose.yml`)

- Renamed the MariaDB container to `wordpress-db`.
- Modified the `WORDPRESS_DB_HOST` environment variable to match the service name with `- WORDPRESS_DB_HOST=wordpress-db`.
- Set the restart policy for both `wordpress` and `wordpress-db` services to `unless-stopped`.

These changes reflect the adjustments made to the configuration files.

steveiliop56 commented on apps/wordpress/config.json:
```diff
@@ -0,0 +1,45 @@
+  "source": "https://github.com/WordPress/WordPress",
+  "website": "https://wordpress.org",
+  "form_fields": [
+    {
+      "type": "random",
+      "label": "Database Name",
+      "min": 32,
+      "env_variable": "DB_NAME"
+    },
```
No need to make the database name random. It would be better to just name it ```wordpress```.

steveiliop56 commented on apps/wordpress/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+  wordpress:
+    container_name: wordpress
+    image: wordpress:6.4.1
+    hostname: wordpress
```
Is the hostname parameter really needed?

steveiliop56 reviewed (changes requested):
Some minor things to fix too.

Mytraas commented:
### Comparison of Configuration Changes

#### Docker Compose YAML File (`docker-compose.yml`)

- Database name is set to `wordpress`.
- Environment variable for `DB_NAME` removed.
- `hostname` removed.

#### JSON Configuration (`config.json`)
- Removed the "Database Name" field from "form_fields" as it's not defined in the Docker Compose file.

steveiliop56 commented on apps/wordpress/config.json:
```diff
@@ -0,0 +1,39 @@
+  "source": "https://github.com/WordPress/WordPress",
+  "website": "https://wordpress.org",
+  "form_fields": [
+    {
+      "type": "random",
+      "label": "Database User",
+      "min": 32,
+      "env_variable": "DB_USER"
+    },
```
That's not really needed just use ```tipi``` as the user.

steveiliop56 reviewed (changes requested):
Hey @Mytraas,

Almost done! For some reason the previous review didn't pick up an extra comment I added so here it is...
